volatility3/framework/plugins/linux/psscan.py

```diff
@@ -0,0 +1,158 @@
+            exit_state,
+        )
+        return task_fields
+
+    def _generator(self):
+        """Generates the tasks found from scanning."""
+
+        for task in self.scan_tasks(
+            self.context, self.config["kernel"], self.config["kernel.layer_name"]
+        ):
+            row = self._get_task_fields(task)
+            yield (0, row)
+
+    @classmethod
+    def scan_tasks(
+        cls,
+        context: interfaces.context.ContextInterface,
+        vmlinux_module_name: str,
+        kernel_layer_name: str,
+    ) -> Iterable[interfaces.objects.ObjectInterface]:
+        """Scans for tasks in the memory layer.
+
+        Args:
+            context: The context to retrieve required elements (layers, symbol tables) from
+            vmlinux_module_name: The name of the kernel module on which to operate
+            kernel_layer_name: The name for the kernel layer
+        Yields:
+            Task objects
+        """
+        vmlinux = context.modules[vmlinux_module_name]
+
+        # check if this image is 32bit or 64bit
+        is_32bit = not symbols.symbol_table_is_64bit(context, vmlinux.symbol_table_name)
+        if is_32bit:
+            pack_format = "I"
+        else:
+            pack_format = "Q"
+
+        # get task_struct to find the offset to the sched_class pointer
+        sched_class_offset = vmlinux.get_type("task_struct").members["sched_class"][0]
+        kernel_layer = context.layers[kernel_layer_name]
+
+        needles = []
+        for symbol in vmlinux.symbols:
+
+            # find all sched_class names by searching by if they include '_sched_class', e.g. 'fair_sched_class'
+            if "_sched_class" in symbol:
+
+                # use canonicalize to set the appropriate sign extension for the addr
+                addr = kernel_layer.canonicalize(vmlinux.get_symbol(symbol).address)
+
+                # append to needles list the packed hex for searching
+                needles.append(struct.pack(pack_format, addr))
+
+        # scan the memory_layer for these needles
+        memory_layer = context.layers["memory_layer"]
+        for address, _ in memory_layer.scan(
+            context, scanners.MultiStringScanner(needles)
+        ):
+            # create task in the memory_layer
+            ptask = context.object(
+                vmlinux.symbol_table_name + constants.BANG + "task_struct",
+                offset=address - sched_class_offset,
+                layer_name="memory_layer",
+            )
+
+            # sanity check exit_state
+            try:
+                # attempt tp parse the exist_state using the enum
+                DescExitStateEnum(ptask.exit_state)
+            except ValueError:
+                vollog.debug(
+                    f"Skipping task_struct at {hex(ptask.vol.offset)} as exit_state {ptask.exit_state} is likely not valid"
+                )
+                continue
+
+            # sanity check pid
+            if not (0 < ptask.pid < 66000):
```

Comment:
Is this checking that pid should be less than 65535?  I know it's just a safety check, but if there's a specific limit, I'd prefer to use that than something that's "close"...